### DISCUSSION TITLE
GCAM 7.0

### DISCUSSION THREAD

**Discussion by pralitp** [JGCRI/gcam-core#295] [Announcements]
New features include the following.  For details see the [updates](http://jgcri.github.io/gcam-doc/updates.html) page on our documentation site.
1. Fix Extrapolation in Residue Biomass
2. Add Land Use History Output
3. Miscellaneous Bug Fixes Part 1
4. Add Exogenous Shutdown Decider
5. Miscellaneous Bug Fixes Part 2
6. GCAM v6.0 transportation bugfix
7. Adding fugitive CO2 emissions from fossil resources
8. AgLU data and method updates (connecting land hectares to food calories)
9. GCAM-USA nonCO2 GHG emissions in industrial and urban processes
10. Detailed Natural Gas Trade
11. [Natural Gas Final Grade Cost and Miscellaneous Solution Improvements
12. Global Iron and Steel Trade
13. GCAM Macro-Economic Module (KLEM Version)
14. gcamdata chunk re-name
15. Update to Hector v3
16. Bugfix before 7.0 Release

<hr /><em>This discussion was created from the release <a href='https://github.com/JGCRI/gcam-core/releases/tag/gcam-v7.0'>GCAM 7.0</a>.</em>

**Comment by ZZHsss0623** [comment]
Is the GCAM-China model in GCAM 7.0? Or do I need to download and install it separately?

**Comment by Thor987** [comment]
Hello, I'm struggling with setting up and running the GCAM SSP1-5 scenarios. Can anyone provide guidance or tips on how to configure these scenarios effectively? Any help you can offer would be greatly appreciated. If possible, could you please share the steps or resources via email at lipengfa12@gmail.com? Thank you in advance for your time and assistance.

**Comment by pkyle** [comment]
If you just want to run SSPs 1-5, with no radiative forcing target, then simply saving `configuration_ssp.xml` as `configuration.xml` and executing the model will work. By default, this configuration file points to `batch_SSP_REF.xml` (in line 5) which runs all 5 SSPs in batch mode. If you want only selected ones of these, you can either delete or comment out sections of that batch file, and if you want to run SSP-RCP combinations, those are available in the various `batch_SSP_SPA*.xml` files.

**Reply by Thor987** [reply]
Thanks to your assistance, I have successfully executed these scenarios. However, I am now interested in running SSP-RCP combinations as well. I noticed in your previous message that SSP-RCP combinations are available in the various batch_SSP_SPA*.xml files. To proceed with running SSP-RCP combinations, could you please clarify whether I should modify the "configuration_ssp.xml" file at line 5 to point to the appropriate batch file (e.g., batch_SSP_SPA*.xml) or if I should modify the line 6, which references the "forcing_target4p5"?

**Comment by pkyle** [comment]
Good question! You can ignore that `forcing_target_4p5.xml` entry in `configuration_ssp.xml` as it won't be used in the set-up that is used for running the SPA files. Instead, the value to change is from `batch_SSP_REF.xml` to `batch_SSP_SPA*.xml`, but please note that the default set-up will run a LOT of scenarios. For example, the file `batch_SSP_SPA1.xml` will try to run 5 different end-of-century targets ranging from 1.9 W/m2 to 6.0 W/m2, and each of these 5 will use "target-finder" to find the GHG emissions pathway that hits the end-of-century target. A typical execution of target-finder may take 5-10 scenario runs to find the path, and the low-emissions ones in particular are prone to model solution failure. In the end, you'd still only get 5 scenarios saved to the model output database, but it could take days to get there on a desktop computer. So, unless you have a lot of computers that you use to keep the house warm in the winter, I would not recommend trying to run these SPA files without first filtering which radiative forcing target levels you actually need, which is done in the last lines of each of those SPA files (`policy-target-runner` section).

**Reply by Thor987** [reply]
Thank you so much for your detailed explanation regarding the setup for running SSP-RCP combinations in GCAM. Your guidance has been immensely helpful in navigating through the complexities of the model.
However, I have noticed that GCAM 7.0 only provides targets ranging from 1.9 W/m2 to 6.0 W/m2, and the targets 7.0 W/m2 and 8.0 W/m2 are missing. Could you please advise me on where I can obtain the "forcing_target_7p0.xml" and "forcing_target_8p5.xml" files? Alternatively, should I attempt to create these files myself based on existing templates?

**Comment by pkyle** [comment]
The reason that the radiative forcing targets only go up to 6.0 (at most) is that most of the reference SSP scenarios (i.e. scenarios with no GHG mitigation policy) don't get a whole lot higher. I think with the technology assumptions in the model right now, emissions factors, and the version of Hector being used, only the SSP5 scenario gets over 7.0 W/m2 by 2100. Please correct me if you see something different in your runs.
Anyways, if you assign target-finder to hit a higher-than-reference emissions target, only bad things can happen from that point, and I hope a model crash happens early in the process, but I don't actually know what will happen.

**Reply by Thor987** [reply]
I wanted to express my sincere gratitude for your recent response regarding the radiative forcing targets in GCAM SSP scenarios.I also wanted to share some observations from my recent runs of the GCAM model.   While running the unmodified "configuration_ssp.xml" file and executing the SSP reference scenarios (SSP1-SSP5), I have noticed that the total climate forcing values for the year 2100 range from 6 to 9 W/m2 across these scenarios.   This observation has raised some questions for me about the accuracy of my runs and whether there may have been errors in the process.

**Reply by pkyle** [reply]
Interesting, thanks for the update! I don't know that anything's wrong in your scenarios; it's been a couple of years since I've looked at forcing across the SSPs. We usually print out `forcing-total` when we're looking at this, and that's what's in the forcing query in `Main_queries.xml`, as well as what target-finder is using in the SPA scenarios. There are other measures of forcing that can be queried, just FYI; probably about 30 total, as all of the F-gases are written specifically. E.g., `forcing-Kyoto`, `forcing-RCP`, `forcing-CO2`

**Reply by ssmithClimate** [reply]
Note that the GCAM SSP scenarios without any forcing constraint will not match the "marker" scenario SSP values, nor will the updated GCAM versions match the forcing from the published GCAM versions that were produced with a previous version of the model. As the model is updated, they will turn out differently as base-year emissions, technology and other drivers are updated.

**Reply by ssmithClimate** [reply]
Note also that there are different forcing definitions as well in terms of what year is taken as the zero point and exactly what species are included in total forcing. Historical forcing is also updated in GCAM as new science becomes available, so this will also cause changes.

**Reply by Thor987** [reply]
Thank you both, pklye and ssmith, for your valuable insights and clarifications regarding the GCAM SSP scenarios and their interpretations.  Your expertise and explanations have provided me with a deeper understanding of the complexities involved in running and interpreting these scenarios.  I will keep your points in mind as I continue my analysis and exploration of the GCAM model.

**Comment by RayRojasCandia** [comment]
Hello everyone, I a newbie using GCAM and I starting to use Linux, so, can GCAM be runned in Linux Mint?

**Reply by MINISUPE** [reply]
您好，我也是新手！期待你的教学

**Comment by nurf004** [comment]
Hello, I noticed that GCAM 7.0 has a new feature for the macroeconomic module. I'm struggling with changing the GDP, considering the Covid-19 shock. In GCAM 6.0, I used an add-on file (population, labor force, and labor productivity) to set my own estimated GDP. However, it didn't work for GCAM 7.0. Should I modify it from the GCAM data system? Or is there any alternative? Thank you in advance for your time and assistance.

**Comment by pkyle** [comment]
I'd recommend first making sure that your configuration.xml file has the following Bool, set to 1 by default:
```
		<Value name="FixedGDP-Path">1</Value>
```
If your configuration doesn't have that, or if it's set to zero, then whatever is assumed for the default trajectory of GDP (generally defined by labor productivity growth rates over time) will determine future global trends.
